x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index/es_index_connector.tsx

```diff
@@ -172,38 +185,40 @@ const IndexActionConnectorFields: React.FunctionComponent<
         }
       />
       <EuiSpacer size="m" />
-      <EuiSwitch
-        data-test-subj="hasTimeFieldCheckbox"
-        checked={hasTimeFieldCheckbox || false}
-        disabled={readOnly}
-        onChange={() => {
-          setTimeFieldCheckboxState(!hasTimeFieldCheckbox);
-          // if changing from checked to not checked (hasTimeField === true),
-          // set time field to null
-          if (hasTimeFieldCheckbox) {
-            editActionConfig('executionTimeField', null);
+      {showTimeFieldCheckbox && (
+        <EuiSwitch
+          data-test-subj="hasTimeFieldCheckbox"
+          checked={hasTimeFieldCheckbox || false}
+          disabled={readOnly}
+          onChange={() => {
+            setHasTimeFieldCheckboxState(!hasTimeFieldCheckbox);
+            // if changing from checked to not checked (hasTimeField === true),
+            // set time field to null
+            if (hasTimeFieldCheckbox) {
+              editActionConfig('executionTimeField', null);
+            }
+          }}
+          label={
+            <>
+              <FormattedMessage
+                id="xpack.triggersActionsUI.components.builtinActionTypes.indexAction.defineTimeFieldLabel"
+                defaultMessage="Define time field for each document"
+              />
+              <EuiIconTip
+                position="right"
+                type="questionInCircle"
+                content={i18n.translate(
+                  'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.definedateFieldTooltip',
+                  {
+                    defaultMessage: `Document ingest time will be written to this field.`,
```

Comment:
@gchaps, not sure about the wording here. Previous tooltip said `Automatically add a time field to each document when it's indexed.` which makes it seem like we would be adding a time field to the document.

What we want to convey is that we will use the field they select to set a timestamp when the document is indexed.

Reply:
How about:

Set the time field to the time the document was indexed.